WordPress/wordpress-develop
# Automatically sync stable blocks from Gutenberg to wordpress-develop on release

## What problem does this PR solve?

Syncing stable blocks from the Gutenberg repository to `wordpress-develop` is currently a manual process.

The thing is, it can be automated.

That's what this PR does.

## How does this PR solve it?

The big idea is to:

1. Read all the `block.json` files from the `@wordpress/block-library` directory.
2. Pick the ones without `__experimental` restrictions – these are the stable blocks.
3. Use that list to generate PHP code that `require`s appropriate dynamic blocks PHP files.
4. Use that list to generate PHP code that calls `remove_action` with appropriate arguments in PHP unit tests.
5. Use that list yet again in Webpack config, instead of the hardcoded one.

All of this is possible thanks to https://github.com/WordPress/gutenberg/pull/40655

To make the process smooth, `require` and `remove_action` calls were extracted to separate PPH files that can be easily autogenerated and overwritten:

* `wp-includes/blocks/require-blocks.php`
* `tests/phpunit/includes/unregister-blocks-hooks.php`

## How to test?

1. Wait for the Gutenberg 13.2 release on May 11th where https://github.com/WordPress/gutenberg/pull/40655 will be included
1. Confirm all the checks are green on this PR
1. Checkout this branch
1. Run `npx grunt sync-gutenberg-packages`
1. Confirm all the stable blocks are now represented in the php files listed above
1. Confirm all the stable blocks were included in the build.
1. Confirm experimental blocks like `post-author-name` were excluded from both lists

cc @gziolo
